runtime/queries/rust/highlights.scm

```diff
@@ -103,72 +50,173 @@
   "<" @punctuation.bracket
   ">" @punctuation.bracket)
 
-"::" @punctuation.delimiter
-"." @punctuation.delimiter
-";" @punctuation.delimiter
+(parameter
+	pattern: (identifier) @variable.parameter)
+(closure_parameters
+	(identifier) @variable.parameter)
 
-(parameter (identifier) @variable.parameter)
-(closure_parameters (_) @variable.parameter)
 
-(lifetime (identifier) @label)
+; Other Global
+
+
+(impl_item
+  "for" @keyword)
+
+"loop" @special
+"for" @special
+"in" @special
+"break" @special
+"continue" @special
+"while" @special
+
+"match" @special
+"if" @special
+"else" @special
+"await" @special
+"return" @special
```

Comment:
These aren't "special". They're keywords. Tag them as `keyword.control` (for control flow), then you can style them separately if you want.

Reply:
See https://www.sublimetext.com/docs/scope_naming.html for scope naming, we deviate a bit and bring a few things to the toplevel iirc